src/govuk/components/pagination/template.test.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-env jest */
+
+const axe = require('../../../../lib/axe-helper')
+
+const { render, getExamples } = require('../../../../lib/jest-helpers')
+
+const examples = getExamples('pagination')
+
+describe('Pagination', () => {
+  describe('Numbered view', () => {
+    it('passes accessibility tests', async () => {
+      const $ = render('pagination', examples.default)
+
+      const results = await axe($.html())
+      expect(results).toHaveNoViolations()
+    })
+
+    it('renders the correct URLs for each link', () => {
+      const $ = render('pagination', examples.default)
+      const previous = $('.govuk-pagination__prev .govuk-pagination__link')
+      const next = $('.govuk-pagination__next .govuk-pagination__link')
+      const firstNumber = $('.govuk-pagination__item:first-child .govuk-pagination__link')
+      const secondNumber = $('.govuk-pagination__item:nth-child(2) .govuk-pagination__link')
+      const thirdNumber = $('.govuk-pagination__item:last-child .govuk-pagination__link')
+
+      expect(previous.attr('href')).toEqual('/previous')
+      expect(next.attr('href')).toEqual('/next')
+      expect(firstNumber.attr('href')).toEqual('/page/1')
+      expect(secondNumber.attr('href')).toEqual('')
+      expect(thirdNumber.attr('href')).toEqual('/page/3')
+    })
+
+    it('renders the correct number within each pagination item', () => {
+      const $ = render('pagination', examples.default)
+      const firstNumber = $('.govuk-pagination__item:first-child')
+      const secondNumber = $('.govuk-pagination__item:nth-child(2)')
+      const thirdNumber = $('.govuk-pagination__item:last-child')
+
+      expect(firstNumber.text().trim()).toEqual('1')
+      expect(secondNumber.text().trim()).toEqual('2')
+      expect(thirdNumber.text().trim()).toEqual('3')
+    })
+
+    // The current item is marked up with a visually hidden span and an aria-hidden span side by side
+    // Instead of the aria-label solution used for the links in the pagination because of issues caused
+    // by aria-label on non-interactive elements like li's
+    it('marks up the current item correctly', () => {
+      const $ = render('pagination', examples.default)
+      const currentNumber = $('.govuk-pagination__item--current')
+      const currentNumberLink = currentNumber.find('.govuk-pagination__link')
+
+      expect(currentNumber).toBeTruthy()
+      expect(currentNumberLink.attr('aria-current')).toEqual('page')
+    })
```

Comment:
Not really sure what we get from testing for `currentNumber` being truthy – if it was falsey then `currentNumberLink` would also have to be falsey and getting the attribute would fail?